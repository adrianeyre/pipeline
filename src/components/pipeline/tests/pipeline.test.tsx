import { render } from '@testing-library/react';
import Pipeline from '../pipeline';
import IPipelineProps from '../interfaces/pipeline-props';

describe('Pipeline Run', () => {
	it('Should render correctly', () => {
		const defaultProps: IPipelineProps = {};
		const pipeline = render(<Pipeline {...defaultProps} />);
		expect(pipeline).toMatchSnapshot();
	});
});