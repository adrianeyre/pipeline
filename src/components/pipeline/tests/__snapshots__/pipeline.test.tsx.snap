// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pipeline Run Should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pipeline-play-container"
        style="max-width: 0px; margin-left: 512px;"
      >
        <div>
          <div
            style="position: absolute; width: 100%; max-width: 0px;"
          >
            <div
              class="info-board"
              style="width: 80%; max-width: 0px;"
            >
              <div
                class="info-board-header"
              >
                <img
                  alt="player"
                  src="player-stand.png"
                />
                <span
                  class="header-text"
                >
                  Pipeline
                </span>
                <img
                  alt="player"
                  src="player-stand.png"
                />
              </div>
              <div
                class="info-board-instructions"
              >
                <p>
                  This is a remake of two of my favourite games for the 
                  <a
                    href="https://en.wikipedia.org/wiki/BBC_Micro"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                     BBC Micro
                  </a>
                  , 
                  <a
                    href="https://en.wikipedia.org/wiki/Repton_(video_game)"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Repton
                  </a>
                   and 
                  <a
                    href="https://en.wikipedia.org/wiki/Pipeline_(video_game)"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Pipeline
                  </a>
                  .
                </p>
                <p>
                  You have to navigate Repton around the playing field collecting all of the stars. There are plenty of puzzles to solve including gates, keys, boulders and teleports
                </p>
                <table>
                  <tbody>
                    <tr>
                      <td
                        class="title"
                      >
                        Function
                      </td>
                      <td
                        class="title"
                      >
                        Key
                      </td>
                    </tr>
                    <tr>
                      <td>
                        Move Up
                      </td>
                      <td>
                        Arrow Up
                      </td>
                    </tr>
                    <tr>
                      <td>
                        Move Down
                      </td>
                      <td>
                        Arrow Down
                      </td>
                    </tr>
                    <tr>
                      <td>
                        Move Left
                      </td>
                      <td>
                        Arrow Left
                      </td>
                    </tr>
                    <tr>
                      <td>
                        Move Right
                      </td>
                      <td>
                        Arrow Right
                      </td>
                    </tr>
                    <tr>
                      <td>
                        Inventory Swap
                      </td>
                      <td>
                        Spacebar
                      </td>
                    </tr>
                    <tr>
                      <td>
                        Inventory Drop
                      </td>
                      <td>
                        Enter
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <div
                class="button-area"
              >
                <button
                  type="button"
                >
                  Play Game
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pipeline-play-container"
      style="max-width: 0px; margin-left: 512px;"
    >
      <div>
        <div
          style="position: absolute; width: 100%; max-width: 0px;"
        >
          <div
            class="info-board"
            style="width: 80%; max-width: 0px;"
          >
            <div
              class="info-board-header"
            >
              <img
                alt="player"
                src="player-stand.png"
              />
              <span
                class="header-text"
              >
                Pipeline
              </span>
              <img
                alt="player"
                src="player-stand.png"
              />
            </div>
            <div
              class="info-board-instructions"
            >
              <p>
                This is a remake of two of my favourite games for the 
                <a
                  href="https://en.wikipedia.org/wiki/BBC_Micro"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                   BBC Micro
                </a>
                , 
                <a
                  href="https://en.wikipedia.org/wiki/Repton_(video_game)"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Repton
                </a>
                 and 
                <a
                  href="https://en.wikipedia.org/wiki/Pipeline_(video_game)"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Pipeline
                </a>
                .
              </p>
              <p>
                You have to navigate Repton around the playing field collecting all of the stars. There are plenty of puzzles to solve including gates, keys, boulders and teleports
              </p>
              <table>
                <tbody>
                  <tr>
                    <td
                      class="title"
                    >
                      Function
                    </td>
                    <td
                      class="title"
                    >
                      Key
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Move Up
                    </td>
                    <td>
                      Arrow Up
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Move Down
                    </td>
                    <td>
                      Arrow Down
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Move Left
                    </td>
                    <td>
                      Arrow Left
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Move Right
                    </td>
                    <td>
                      Arrow Right
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Inventory Swap
                    </td>
                    <td>
                      Spacebar
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Inventory Drop
                    </td>
                    <td>
                      Enter
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div
              class="button-area"
            >
              <button
                type="button"
              >
                Play Game
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
